### PR TITLE
Update check

### PR DESCRIPTION
**Pull request by r2d4**
* uploads a VERSION file to latest bucket
* non blocking check for updates on every command, i.e. command will not wait for version check to happen before exiting
* if theres an update, it prints out a message at the very end of output
* won't check on dirty trees - the way that we inject the version in the makefile makes it so that if you don't explicitly pull tags you'll be left with an "old" version.

merge this after this release

### CODE REVIEW

**Comment by r2d4** [comment]
Fixes https://github.com/GoogleContainerTools/skaffold/issues/148

File: cmd/skaffold/app/cmd/cmd.go [hunk 1]
```diff
@@ -64,6 +82,25 @@ func NewSkaffoldCommand(out, err io.Writer) *cobra.Command {
...
+	current, err := version.ParseVersion(version.Get().Version)
+	if err != nil {
+		return errors.Wrap(err, "parsing current semver, skipping update check")
+	}
+	latest, err := update.GetLatestVersion(context.Background())
+	if err != nil {
+		return errors.Wrap(err, "getting latest version")
+	}
+	if latest.GT(current) {
+		ch <- fmt.Sprintf("There is a new version (%s) of skaffold available.", latest)
```
Comment (balopat) [inline cmd/skaffold/app/cmd/cmd.go]: nit: provide a download link

Comment (r2d4) [reply cmd/skaffold/app/cmd/cmd.go]: good idea, done

**Comment by r2d4** [comment]
We should also probably tell users how to disable this by setting `SKAFFOLD_UPDATE_CHECK=false`

**Pull request merged**